exercises/05.actions/05.solution.history-revalidation/server/app.js

```diff
@@ -1,66 +1,72 @@
-import bodyParser from 'body-parser'
-import busboy from 'busboy'
+import { readFile } from 'fs/promises'
+import { serve } from '@hono/node-server'
+import { serveStatic } from '@hono/node-server/serve-static'
+import { RESPONSE_ALREADY_SENT } from '@hono/node-server/utils/response'
 import closeWithGrace from 'close-with-grace'
-import compress from 'compression'
-import express from 'express'
+import { Hono } from 'hono'
+import { trimTrailingSlash } from 'hono/trailing-slash'
 import { createElement as h } from 'react'
 import {
 	renderToPipeableStream,
-	decodeReplyFromBusboy,
+	decodeReply,
 } from 'react-server-dom-esm/server'
 import { App } from '../src/app.js'
 import { shipDataStorage } from './async-storage.js'
 
 const PORT = process.env.PORT || 3000
 
-const app = express()
-app.use(compress())
-// this is here so the workshop app knows when the server has started
-app.head('/', (req, res) => res.status(200).end())
+const app = new Hono({ strict: true })
 
-app.use(express.static('public', { index: false }))
-app.use('/js/src', express.static('src'))
+app.use(trimTrailingSlash())
 
-// This just cleans up the URL if the search ever gets cleared... Not important
-// for RSCs... Just ... I just can't help myself. I like URLs clean.
-app.use((req, res, next) => {
-	if (req.query.search === '') {
-		const searchParams = new URLSearchParams(req.search)
-		searchParams.delete('search')
-		const location = [req.path, searchParams.toString()]
-			.filter(Boolean)
-			.join('?')
-		return res.redirect(302, location)
-	} else {
-		next()
-	}
-})
```

Comment:
This use case is not handled. If I go to `/bc4cbadf89bd3?search=`, it should redirect me to `/bc4cbadf89bd3`, but right now this PR gives a 404.